flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultExecutionSlotAllocatorTest.java

```diff
@@ -216,13 +221,37 @@ private SlotExecutionVertexAssignment findSlotAssignmentByExecutionVertexId(
 				Time timeout) {
 
 			slotAllocationRequests.add(Tuple3.of(slotRequestId, task, slotProfile));
-			if (slotAllocationDisabled) {
+			if (forceFailingSlotAllocation) {
+				return FutureUtils.completedExceptionally(new Exception("Forced failure"));
+			} else if (slotAllocationDisabled) {
 				return new CompletableFuture<>();
 			} else {
 				return CompletableFuture.completedFuture(new TestingLogicalSlotBuilder().createTestingLogicalSlot());
 			}
 		}
 
+		public CompletableFuture<Collection<PhysicalSlotRequest.Result>> allocatePhysicalSlots(
```

Comment:
If we would manage to make `OneSlotPerExecutionSlotAllocator` depending on `BulkSlotProvider`, this `allocatePhysicalSlots` could in a separate `AllocationToggableBulkSlotProvider` or so, as a separate commit if we do it.

Reply:
Yes, `OneSlotPerExecutionSlotAllocatorTest` is reworked and does not use `AllocationToggableBulkSlotProvider` anymore.